granulate_utils/linux/process.py

```diff
@@ -14,8 +20,13 @@ def process_exe(process: psutil.Process) -> str:
     See https://github.com/giampaolo/psutil/pull/2062
     """
     exe = process.exe()
-    if exe == "" and is_process_zombie(process):
-        raise psutil.ZombieProcess(process.pid)
+    if exe == "":
+        if is_process_zombie(process):
+            raise psutil.ZombieProcess(process.pid)
+        # following to https://man7.org/linux/man-pages/man5/proc.5.html,
+        # an empty exe can be returned in case of a broken link
```

Comment:
> If the pathname has been unlinked, the symbolic link will contain the string '(deleted)' appended to the original pathname.

That's what I'm seeing.

Also seeing this:

> In a multithreaded process, the contents of this symbolic link are not available if the main thread has already terminated

Which might be something that we encounter